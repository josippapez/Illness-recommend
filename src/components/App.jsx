import React from 'react';
import './App.css';
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';
import PrivateRoute from './PrivateRoute';
import Login from './Login/Login';
import Routes from './Routes';

function App() {
  return (
    <Router>
      <Switch>
        <Route path="/login" component={Login} />
        <PrivateRoute component={Routes} />
      </Switch>
    </Router>
  );
}

export default App;
