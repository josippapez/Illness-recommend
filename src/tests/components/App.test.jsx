/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import App from '../../components/App';

const initialState = {
  user: {
    userInfo: {
      data: null,
      error: null,
      status: null,
    },
    role: null,
    isAuthenticated: false,
    error: null,
    status: null,
    message: null,
  },
  usersList: {
    users: null,
    error: null,
    status: null,
  },
  alergies: {
    alergies: null,
    error: null,
    status: null,
    message: null,
  },
  medicationList: {
    medicationInfo: {
      data: null,
      error: null,
      status: null,
    },
    medications: null,
    error: null,
    status: null,
  },
  symptoms: {
    symptoms: null,
    error: null,
    status: null,
    message: null,
  },
  theme: {
    darkTheme: false,
  },
  patient: {
    currentPatientInfo: null,
    viewingPatientInfo: null,
    error: null,
    status: null,
    message: null,
    patientList: null,
  },
};

const mockStore = configureStore();
const store = mockStore(initialState);

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect,
}));

const component = shallow(
  <Provider store={store}>
    <App />
  </Provider>
);

describe('App', () => {
  it('renders correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
