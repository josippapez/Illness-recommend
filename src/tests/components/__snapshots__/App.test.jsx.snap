// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `"<div class=\\"navigation-setup\\"><div class=\\"Toastify\\"></div></div>"`;
